src/Microsoft.AspNet.Mvc.Razor/RazorTextWriter.cs

```diff
@@ -30,9 +32,12 @@ public class RazorTextWriter : TextWriter, IBufferedTextWriter
         /// <param name="unbufferedWriter">The <see cref="TextWriter"/> to write output to when this instance
         /// is no longer buffering.</param>
         /// <param name="encoding">The character <see cref="Encoding"/> in which the output is written.</param>
-        public RazorTextWriter(TextWriter unbufferedWriter, Encoding encoding)
+        /// <param name="encoder">The HTML encoder.</param>
+        public RazorTextWriter(TextWriter unbufferedWriter, Encoding encoding, IHtmlEncoder encoder)
```

Comment:
Changes to `StringCollectionTextWriter` mean `Write(string)` does not result in encoding. Changes to this class are about moving from property to constructor injection for the `HtmlEncoder`.